proposed/2023/package-source-mapping-actions-PMUI.md

```diff
@@ -0,0 +1,102 @@
+# Package Source Mapping Package Management Actions in the PM UI 
+* Start Date: 6/25/22
+* Authors: Ella McNally, Donnie Goodson ([donnie-msft](https://github.com/donnie-msft))
+* Issue: https://github.com/NuGet/Home/issues/11797
+* Status: Draft
+
+## Summary
+
+Currently, Install/Update Package Management actions in the NuGet Package Manager in Visual Studio (PM UI) do not support creating new package source mappings. We want to allow customers to continue using Package Source Mapping as they use the PM UI to manage their packages. 
+
+Package Source Mapping in Restore was introduced in: https://github.com/NuGet/Home/blob/dev/implemented/2021/PackageSourceMapping.md
+
+Package Source Mapping Status in the PM UI was introduced in: [Package Source Mapping Package Management Status in the PM UI ](package-source-mapping-status-PMUI.md)
+
+## Motivation
+
+Adding support for package source mapping actions in PM UI will allow customers to more seamlessly onboard and manage package source mappings while continuing to use the actions they are used to in the PM UI. It also increases awareness of where their packages will be coming from before initiating a package management action.
+
+## Explanation
+
+### Functional Explanation
+
+When a new package source mapping is being created, the Preview Window will appear showing which packages (the top-level and any transitive dependencies) will be mapped. 
+
+(TBD) It is not yet understood whether there will be a new afforance on the details pane in a new row below the Install/Update buttons, or if source mappings will be handled in the background.
+
+#### A Proposal for Consent to a new Source Mapping
+ A `CheckBox` that allows the customer to choose if they want to add a mapping to the package from the selected source when they `Install`/`Update`. This `CheckBox` will always be shown even if package source mapping is not enabled. Package source mapping can be enabled by the customer checking the `CheckBox` to create the first mapping for the solution's settings. Once package source mapping is enabled, the customer must either have a mapping previously configured, or consent to a new mapping by checking the `CheckBox` to install. Otherwise, Restore will fail as it does today.
```

Comment:
Is there a screenshot that we can see?

TBH, I feel like *if* you have source mapping enabled, and a package is not mapped, the mapping additon should be the default by whatever means you opt into it.